### PR TITLE
fix: home screen flickers while creating new workspace

### PR DESCRIPTION
**Pull request by itsmdasifraza**
## Description
fix home screen flickers while creating new workspace
Fixes issue #315 
Fixes issue #286 

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build

## Have you tested locally?

- [x] 👍 yes
- [ ] 🙅 no, because I am lazy

### CODE REVIEW

File: src/lib/components/header/header-dashboard/HeaderDashboard.ViewModel.ts [hunk 1]
```diff
@@ -154,11 +153,7 @@ export class HeaderDashboardViewModel {
     const response = await this.workspaceService.fetchWorkspaces(userId);
     let isAnyWorkspaceActive: undefined | string = undefined;
     const data = [];
-    if (
-      response?.isSuccessful &&
-      response?.data?.data &&
-      response?.data?.data?.length > 0
-    ) {
+    if (response?.isSuccessful && response?.data?.data) {
```
Comment (parthdalaltech) [inline src/lib/components/header/header-dashboard/HeaderDashboard.ViewModel.ts:156]: destructure this beforehand and then use it in if statement like
const {isSuccessful,data} = response;

File: src/lib/components/header/header-dashboard/HeaderDashboardDropdown.svelte [hunk 2]
```diff
@@ -112,6 +117,50 @@
...
+    let workspaces = allworkspaces
+      .filter((elem) => {
+        if (currentTeam.id === elem.team.teamId) return true;
+        return false;
+      })
+      .reverse()
+      .slice(0, workspaceLimit)
+      .map((workspace) => {
+        const workspaceObj = {
+          id: workspace.name,
```
Comment (parthdalaltech) [inline src/lib/components/header/header-dashboard/HeaderDashboardDropdown.svelte:142]: dont we have "id" key of workspace? coz i can see we are assigning id as the name key

**Review by parthdalaltech — CHANGES_REQUESTED** [review]

**Review by LordNayan — APPROVED** [review]

**Pull request merged**